### PR TITLE
[3.7] TSB and ASB should use openshift_hosted_infra_selector by default

### PR DESCRIPTION
**Pull request by vrutkovs**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1538616

This seems to be already implemented in 3.9+ so no need to cherrypick

### CODE REVIEW

File: roles/openshift_service_catalog/defaults/main.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
 openshift_service_catalog_remove: false
-openshift_service_catalog_nodeselector: {"openshift-infra": "apiserver"}
+openshift_service_catalog_nodeselector: "{{ openshift_hosted_infra_selector | default('openshift-infra=apiserver') | map_from_pairs }}"
```
Comment (sdodson) [inline roles/openshift_service_catalog/defaults/main.yml]: Ugh, where'd this `openshift-infra=apiserver` come from? SC seems to be the only place that we're expecting that node selector.

**Comment by sdodson** [comment]
/lgtm cancel
want to look more into mike's comments

File: roles/openshift_service_catalog/defaults/main.yml (hunk 1 above)
Comment (sdodson) [reply roles/openshift_service_catalog/defaults/main.yml]: Oh well, i think the fix is fine. I'd remove the additional defaults in the tasks though.

**Comment by vrutkovs** [comment]
>Is any of this even desirable? I don't know the constraints for 3.7 and service catalog, but this would result presumably in only one node being scheduable by default for SC.

Right, SC change needs to be reverted. This would affect TSB only

**Comment by vrutkovs** [comment]
bot, retest this please

**Pull request merged**